src/Groups/GrpAb.jl

```diff
@@ -12,3 +12,348 @@ function restrict_codomain(f::GrpAbFinGenMap)
+
+function comm(x::GrpAbFinGenElem, y::GrpAbFinGenElem)
+  x.parent == y.parent || error("Elements must belong to the same group")
+  zero(parent(x))
+end
+
+has_gens(G::GrpAbFinGen) = true
+
+#TODO: improve?
+small_generating_set(G::GrpAbFinGen) = gens(G)
+
+
+################################################################################
+#
+#   Conjugacy Classes in GrpAbFinGen groups: just wrap elements
+#
+################################################################################
+
+struct GrpAbFinGenConjClass{T<:GrpAbFinGen, S<:Union{GrpAbFinGenElem,GrpAbFinGen}} <: GroupConjClass{T, S}
+   X::T
+   repr::S
+end
+
+function Base.hash(C::GrpAbFinGenConjClass, h::UInt)
+  return Base.hash(Representative(C), h)
+end
+
+function Base.show(io::IO, C::GrpAbFinGenConjClass)
+  print(io, string(representative(C)), " ^ ", string(C.X))
+end
+
+==(a::GrpAbFinGenConjClass, b::GrpAbFinGenConjClass) = representative(a) == representative(b)
+
+Base.length(::Type{T}, C::GrpAbFinGenConjClass) where T <: IntegerUnion = T(1)
+
+Base.length(C::GrpAbFinGenConjClass) = ZZRingElem(1)
+
+
+################################################################################
+#
+#   Conjugacy classes of elements
+#
+################################################################################
+
+conjugacy_class(G::GrpAbFinGen, g::GrpAbFinGenElem) = GrpAbFinGenConjClass(G, g)
+
+Base.eltype(::Type{GrpAbFinGenConjClass{T,S}}) where {T,S} = S
+
+Base.rand(C::GrpAbFinGenConjClass) = representative(C)
+
+Base.rand(rng::Random.AbstractRNG, C::GrpAbFinGenConjClass) = representative(C)
+
+number_conjugacy_classes(G::GrpAbFinGen) = ZZRingElem(order(G))
+
+number_conjugacy_classes(::Type{T}, G::GrpAbFinGen) where T <: IntegerUnion = T(order(G))
+
+conjugacy_classes(G::GrpAbFinGen) = [GrpAbFinGenConjClass(G, x) for x in G]
+
+is_conjugate(G::GrpAbFinGen, x::GrpAbFinGenElem, y::GrpAbFinGenElem) = (x == y)
+
+function representative_action(G::GrpAbFinGen, x::GrpAbFinGenElem, y::GrpAbFinGenElem)
+   x == y ? (true, zero(G)) : (false, nothing)
+end
+
+
+################################################################################
+#
+#   Conjugacy classes of subgroups
+#
+################################################################################
+
+conjugacy_class(G::T, H::T) where T <: GrpAbFinGen = GrpAbFinGenConjClass(G, H)
+
+function conjugacy_classes_subgroups(G::GrpAbFinGen)
+   return [conjugacy_class(G, H) for (H, mp) in subgroups(G)]
+end
+
+#TODO: Does Hecke provide the following?
+# subgroup_reps(G::GrpAbFinGen; order::ZZRingElem = ZZRingElem(-1))
+# conjugacy_classes_maximal_subgroups(G::GrpAbFinGen)
+# maximal_subgroup_reps(G::GrpAbFinGen)
+# low_index_subgroup_reps(G::GrpAbFinGen, n::Int)
+
+conjugate_group(G::GrpAbFinGen, x::GrpAbFinGenElem) = G
+
+Base.:^(G::GrpAbFinGen, x::GrpAbFinGenElem) = G
+
+function is_conjugate(G::GrpAbFinGen, H::GrpAbFinGen, K::GrpAbFinGen)
+   if is_subgroup(H, K)[1] && is_subgroup(K, H)[1]
```

Comment:
Should this also have a test whether `H` and `K` are actually subgroups of `G`?

Out of curiosity, does `H == K` also work? With all those magic embeddings, this is not clear to me.

Reply:
No, `H == K` is not implied by `is_subgroup(H, K)[1] && is_subgroup(K, H)[1]`.

Reply:
For group theoretic questions, it is usual to construct subgroups in different ways and then ask whether they are equal. From this point of view, the setup for `GrpAbFinGen` is indeed strange.
Here is an example.
```
julia> g = abelian_group([2,3]);  # cyclic group of order 6

julia> s1, emb1 = sylow_subgroup(g, 3);  # the unique subgroup of order 3

julia> s2 = low_index_subgroup_reps(g, 2)[2]; # another way to get the subgroup of order 3

julia> flag2, emb2 = is_subgroup(s2, g); flag2
true

julia> order(s1) == order(s2)
true

julia> is_subgroup(s1, s2)[1] && is_subgroup(s2, s1)[1]  # we can check containment ("magically", w.r.t. the embeddings into g)
true

julia> s1 == s2  # the two groups are not regarded as equal
false
```
The point seems to be that `s1` and `s2` are regarded as independent *groups*, and that only `emb1` and `emb2` interpret them as *subgroups* of `g`. Perhaps the cleanest way to ask for the equality as subgroups of `g` would be something like `is_equal_image(emb1, emb2)`.